scripts/check-duplication.sh

```
@@ -0,0 +1,43 @@
+# Check for duplicate test cases
+echo "🧪 Checking for duplicate test cases..."
+DUPLICATE_TESTS=$(find frontend/src -type f \( -name "*.test.ts" -o -name "*.test.tsx" \) -exec grep -oh "it(['\"][^'\"]*['\"]" {} \; | sort | uniq -d || true)
+if [ ! -z "$DUPLICATE_TESTS" ]; then
+    echo "❌ Found potential duplicate test cases:"
+    echo "$DUPLICATE_TESTS"
+    exit 1
+fi
+
+# Check for duplicate function names
+echo "🔧 Checking for duplicate function names..."
+DUPLICATE_FUNCTIONS=$(find frontend/src -name "*.ts" -o -name "*.tsx" | xargs grep -h "function\|const.*=" | grep -o "function [a-zA-Z_][a-zA-Z0-9_]*\|const [a-zA-Z_][a-zA-Z0-9_]*" | sort | uniq -d || true)
```

Shellcheck (0.11.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)